File: daimond pattern/daimondpattern.py
def erect_pyramid(N):
    for i in range(N):
        row = " " * (N - i - 1) + "*" * (2 * i + 1) + " " * (N - i - 1)
        print(row)

def inverted_pyramid(N):
    for i in range(N):
        row = " " * i + "*" * (2 * N - (2 * i + 1)) + " " * i
        print(row)

def main():
    N = 5
    erect_pyramid(N)
    inverted_pyramid(N)

if __name__ == "__main__":
    main()

File: daimond pattern/daimondpattern.js
function erect_pyramid(N) {
    for (let i = 0; i < N; i++) {
        let row = "";
        for (let j = 0; j < N - i - 1; j++) {
            row += " ";
        }
        for (let j = 0; j < 2 * i + 1; j++) {
            row += "*";
        }
        for (let j = 0; j < N - i - 1; j++) {
            row += " ";
        }
        console.log(row);
    }
}

function inverted_pyramid(N) {
    for (let i = 0; i < N; i++) {
        let row = "";
        for (let j = 0; j < i; j++) {
            row += " ";
        }
        for (let j = 0; j < 2 * N - (2 * i + 1); j++) {
            row += "*";
        }
        for (let j = 0; j < i; j++) {
            row += " ";
        }
        console.log(row);
    }
}

function main() {
    const N = 5;
    erect_pyramid(N);
    inverted_pyramid(N);
}

main();
